### DISCUSSION TITLE
About "RuntimeWarning: invalid value in power ..." warnings.

### DISCUSSION THREAD

**Discussion by onlyhope10** [colour-science/colour#727] [Q&A] [answered]
xdata=np.array([10,20,30,60,120,180,360,720,1440])
ydata=np.array([243,194,171.9,128.1,76.7,54.7,31,16.2,8.4])

def func(xdata,a,b,n):
    return a/(xdata+b)**n

init=np.array([1,1,1])

best_val,covar= curve_fit(func,xdata,ydata,p0=init)
print(best_val)

Warning (from warnings module):
  File "G:\least_sq.py", line 12
    return a/(xdata+b)**n
RuntimeWarning: invalid value encountered in power
[1.14275303e+04 4.08207279e+01 9.84057243e-01]

Please help me how do I fix my RuntimeWarning..thanks

**Comment by thomasmansencal** [comment, marked as answer]
Hi @onlyhope10,

This code has nothing to do with [Colour](https://github.com/colour-science/colour) so I don't think this issue belongs here. That said what you are getting here is a warning that happened when **Scipy** tested values in the optimisation process and probably ended up raising a negative number to a fractional power or something along those lines.

**Comment by thomasmansencal** [comment]
Closing this one, feel free to continue the discussion though!
